clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -164,13 +166,23 @@ public ConsumerCoordinator(GroupRebalanceConfig rebalanceConfig,
               metricGrpPrefix,
               time);
         this.rebalanceConfig = rebalanceConfig;
-        this.log = logContext.logger(ConsumerCoordinator.class);
+        final Supplier<String> dynamicPrefix =
+            () -> logContext.logPrefix() + "[generationId=" + generation().generationId + "] ";
+        this.log = new DynamicPrefixLogger(
+            dynamicPrefix,
+            LoggerFactory.getLogger(ConsumerCoordinator.class)
+        );
```

Comment:
Is it worth trying to modify the logger used by the superclass as well?

Reply:
If we don't want to modify `AbstractCoordinator` to include generation IDs for all of its subclasses instead of just `ConsumerCoordinator`, we could modify `AbstractCoordinator` to also implement `ContextualLogging` and then invoke `super::setLoggingContext` in the `ConsumerCoordinator` constructor.

Reply:
Thanks, @C0urante !

I think for now, I'd like to keep the change on the client side of the protocol only. The main intent of this change is to make it easier to see what's happening with rebalances on the client side, and I'm also a little concerned about the performance impact of this code on the broker.

Perf is not as concerning on the client side because a single consumer group spends most of its time processing and only occasionally performs a rebalance. In contrast, a broker managing a lot of consumer groups may spend a lot more time logging these messages.

Reply:
Thanks @vvcephei--I'm personally a little skeptical that one stack frame is going to have a large performance impact for logging, even broker-side, but I don't mind erring on the side of caution here for now.

I do still wonder about modifying the logger used by the `AbstractCoordinator` superclass, since that's (AFAIK) only used client-side?